pkg/api/controller.go

```diff
@@ -3635,6 +3636,57 @@ func (c *Controller) GetLakeFSVersion(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
+func (c *Controller) ReportUsageEvent(w http.ResponseWriter, r *http.Request, body ReportUsageEventJSONRequestBody) {
+	ctx := r.Context()
+	user, err := auth.GetUser(ctx)
+	if err != nil {
+		writeError(w, r, http.StatusUnauthorized, ErrAuthenticatingRequest)
+		return
+	}
+
+	usageClass, usageName, usageCount := body.Class, body.Name, body.Count
+	if usageClass == "" {
+		writeError(w, r, http.StatusBadRequest, "invalid usage class: cannot use an empty string")
+		return
+	}
+
+	if usageName == "" {
+		writeError(w, r, http.StatusBadRequest, "invalid usage name: cannot use an empty string")
+		return
+	}
+
+	if usageCount <= 0 {
+		writeError(w, r, http.StatusBadRequest, fmt.Sprintf("invalid usage count=%d", usageCount))
+		return
+	}
+
+	client := httputil.GetRequestLakeFSClient(r)
+	ev := stats.Event{
+		Class:      usageClass,
+		Name:       usageName,
+		Repository: "",
+		Ref:        "",
+		SourceRef:  "",
+		UserID:     user.Username,
+		Client:     client,
+	}
+	for i := 0; i < body.Count; i++ {
+		c.Collector.CollectEvent(ev)
+	}
```

Comment:
This will be very inefficient.  Can we change the collector interface to allow counts?

Reply:
done